agent/agent.go

```diff
@@ -2551,7 +2577,7 @@ func (a *Agent) addProxyLocked(proxy *structs.ConnectManagedProxy, persist, From
 		}
 	}
 
-	err = a.addServiceLocked(proxyService, chkTypes, persist, token, source)
+	err = a.addServiceLocked(proxyService, chkTypes, persist, token, true, source)
```

Comment:
Why do you pass ` true` here? Shouldn't it be `false` which is the current behaviour?